### PR TITLE
feat: Added Transpose to Mathematical Funcations

### LINKED ISSUE #25811: <transpose>

### PR DESCRIPTION
**Pull request by thelonewolff**
# PR Description

## Related Issue

Close #25811

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by tomatillos** [comment]
Hi, would you mind writing a test for this function as well? Then we can get it merged :) Thanks

**Comment by thelonewolff** [comment]
Hi, I have now added the test for this function. Could you please review it? Thanks.

**Comment by thelonewolff** [comment]
Hi, I have updated the codes in the function and test, as well as renamed the test to test_jax_transpose.
Could you please review it again?

**Comment by thelonewolff** [comment]
Hi, the "Display_test_results" is showing as passed, so does this mean that it is okay or anything else needs to be added as well?

File: ivy/functional/frontends/jax/numpy/mathematical_functions.py [hunk 1]
```diff
@@ -708,6 +709,11 @@ def trace(a, offset=0, axis1=0, axis2=1, out=None):
     return ivy.trace(a, offset=offset, axis1=axis1, axis2=axis2, out=out)
 
 
+@to_ivy_arrays_and_back
+def transpose(a, axes=None):
+    return mp_routines.transpose_like_operations.transpose(a, axes=axes)
```
Comment (tomatillos) [inline ivy/functional/frontends/jax/numpy/mathematical_functions.py:714]: I think we should use ivy.transpose instead here

Comment (tomatillos) [reply ivy/functional/frontends/jax/numpy/mathematical_functions.py:714]: Ah sorry, use ivy.permute_dims

Comment (thelonewolff) [reply ivy/functional/frontends/jax/numpy/mathematical_functions.py]: Sure, Updated the code with ivy.permute_dims

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_mathematical_functions.py [hunk 2]
```diff
@@ -3218,6 +3218,38 @@ def test_jax_trace(
...
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        backend_to_test=backend_fw,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        test_values=False,
+        a=x[0],
+        axes=None,
```
Comment (tomatillos) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_mathematical_functions.py:3249]: Please test the axis as well

Comment (thelonewolff) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_mathematical_functions.py]: Added the test for axis

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_mathematical_functions.py [hunk 3]
```diff
@@ -3218,6 +3218,38 @@ def test_jax_trace(
...
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        backend_to_test=backend_fw,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        test_values=False,
+        a=x[0],
+        axes=ivy.arange(x[0].ndim),
```
Comment (tomatillos) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_mathematical_functions.py:3249]: Now you're just testing transpose with no permutations. You could look at how we test permute/transpose in the other frameworks for guidance.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Pull request closed without merging**